### PR TITLE
Fix/tao 10346/remove unused delivery param

### PR DESCRIPTION
**Pull request by jbout**
getResultStorage() does not require a parameter so no use in retrieving delivery execution

### CODE REVIEW

File: model/execution/DeliveryServerService.php [hunk 1]
```diff
@@ -109,19 +109,17 @@ public function getDeliveryContainer(DeliveryExecution $deliveryExecution)
     }
 
     /**
-     * @param $deliveryExecution
+     * @param string $deliveryExecution
```
Comment (gitromba) [inline model/execution/DeliveryServerService.php:112]: Is it an ID and or sth else? Below it has been renamed to Id.

File: model/execution/DeliveryServerService.php [hunk 2]
```diff
@@ -109,19 +109,17 @@ public function getDeliveryContainer(DeliveryExecution $deliveryExecution)
...
      * @return ResultStorageWrapper
      * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
      */
-    public function getResultStoreWrapper($deliveryExecution)
+    public function getResultStoreWrapper($deliveryExecutionId)
```
Comment (gitromba) [inline model/execution/DeliveryServerService.php:116]: According to the method description above you should enforce explicitly the `string` type.

File: model/execution/DeliveryServerService.php [hunk 3]
```diff
@@ -109,19 +109,17 @@ public function getDeliveryContainer(DeliveryExecution $deliveryExecution)
...
     {
-        //@todo check if cache should be used here
-        if (!$deliveryExecution instanceof DeliveryExecutionInterface) {
-            $deliveryExecution = ServiceProxy::singleton()->getDeliveryExecution($deliveryExecution);
+        if ($deliveryExecutionId instanceof DeliveryExecutionInterface) {
```
Comment (gitromba) [inline model/execution/DeliveryServerService.php:118]: Now I'm wondering which one is true? Is this is a string? Or an ID represented as an object (a value objec?) or other domain entity?

Comment (jbout) [reply model/execution/DeliveryServerService.php:118]: I understand your confusion, let me clarify:
* This function expected a delivery execution in the past, but accepted an id and did the lookup itself (requires a database lookup)
* After this fix the function expects a string (this is how it is called in all but 1 instances) but in order to not break backward compatibility it still accepts delivery executions, and retrieves the id from them (does not require a database lookup)

File: model/execution/DeliveryServerService.php (hunk 1 above)
Comment (jbout) [reply model/execution/DeliveryServerService.php]: fixed

File: model/execution/DeliveryServerService.php (hunk 3 above)
Comment (gitromba) [reply model/execution/DeliveryServerService.php:116]: You made it clear, thanks!

Now that I understood, there's an old contract should be still supported along with a new one, I would rather introduce the new one in parallel, ideally like:
```
getResultStoreWrapper(DeliveryExecutionInterface  $deliveryExecution): ResultStorageWrapper
getResultStoreWrapper(string  $deliveryExecutionId): ResultStorageWrapper
```
and in PHP
```
getResultStoreWrapper(DeliveryExecutionInterface  $deliveryExecution): ResultStorageWrapper
getResultStoreWrapperByDeliveryExecutionId(string  $deliveryExecutionId): ResultStorageWrapper
```

Comment (jbout) [reply model/execution/DeliveryServerService.php:116]: Unfortunately of the 8 times it is called in taoQtiTest right now it is called 7 times with an identifier and once with a delivery execution.
By splitting it into two different functions that are each strongly typed I would still break backward compatibility, in which case I would rather drop the signature supporting the delivery execution call, as I am providing a PR to change that one in the near future as well.

Comment (gitromba) [reply model/execution/DeliveryServerService.php:116]: In the light of new information I would also say let's play the safe game in this PR. Refactoring will be carried out in a separate PR.

Comment (gitromba) [reply model/execution/DeliveryServerService.php:116]: - [] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [] New code is respecting code style rules
- [] New code is respecting best practices
- [] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [] Acceptance criteria are respected
- [] Pull request title and description are meaningful
- [] Changelog is updated according to changes (if applicable)
- [] Documentation is updated according to changes (if applicable)

**Pull request merged**